### ISSUE TITLE
hlint suggestions are removed from vscode problems tab almost immediately

### ISSUE THREAD

**Issue by jneira** [haskell/haskell-ide-engine#1441]
Hi, when using master hie opening a haskell file in which `hlint <file>` shows hints i get to see them in the vscode problems tab but they are removed almost immediately. 

When using the hie-bios branch the suggestions arent shown at all (or too fast for my eyes :wink:)

Tested in windows 7 and 10.

In linux the behaviour is correct, in master and hie-bios branches.

**Comment by jneira** [comment]
With hie-bios branch, i've  got several errors alerts in vscode with the message `lintCmd: no access to the persisted file.` but it seems it is not related with the issue (see https://github.com/mpickering/haskell-ide-engine/issues/36#issuecomment-551730329)

**Comment by alanz** [comment]
In the past, we had hlint hints flickering from the following scenario

a. file is saved, submitted for diagnostics for both hlint and ghc
b. hlint returns quickly, we send the hints as diagnostics
c. ghc returns later, reporting compilation errors in the saved file.
d. We choose not to show hints if there are compilation errors, so they are flushed and the ghc diagnostics only are sent.

It is possible that this is happening now, but step `c` is returning some other error, which does not result in diagnostics being sent.

**Comment by jneira** [comment]
I've reproduced ther error and i've observed in the [log](https://gist.github.com/jneira/8e3f9c669daeb7b12ca943211d08364f) that behaviour, but the compilation errors are empty (and there are no other messages between):
```
2019-11-12 00:47:50.9707483 [ThreadId 5] - <--2--{"jsonrpc":"2.0","params":{"uri":"file:///d:/dev/ws/haskell/stack-test/test/MyLibTest.hs","diagnostics":[{"severity":3,"range":{"start":{"line":3,"character":7},"end":{"line":4,"character":46}},"code":"Redundant do","source":"hlint","message":"Redundant do\nFound:\n  do putStrLn \"Test suite not yet implemented.\"\nWhy not:\n  putStrLn \"Test suite not yet implemented.\"\n"}]},"method":"textDocument/publishDiagnostics"}
2019-11-12 00:48:03.1326663 [ThreadId 11] - setTypecheckedModule: after ghc-mod
2019-11-12 00:48:03.1336657 [ThreadId 11] - setTypecheckedModule: Did get typechecked module for: "d:\\dev\\ws\\haskell\\stack-test\\test\\MyLibTest.hs"
2019-11-12 00:48:03.1376596 [ThreadId 11] - setTypecheckedModule: done
2019-11-12 00:48:03.1376596 [ThreadId 11] - ghcDispatcher: top of loop
2019-11-12 00:48:03.1376596 [ThreadId 5] - <--2--{"jsonrpc":"2.0","params":{"uri":"file:///D:/dev/ws/haskell/stack-test/test/MyLibTest.hs","diagnostics":[]},"method":"textDocument/publishDiagnostics"}
```

**Comment by fendor** [comment]
This has been closed by updating haskell-lsp, right?

**Comment by jneira** [comment]
As #1447 the change it is not included in haskell-lsp-0.18 but 0.19 so it can only be included in hie-bios.
Nevertheless the issue will be fixed in the incoming hie-bios branch so we could keep this close

**Comment by fendor** [comment]
thanks for the clarification!
